### ISSUE TITLE
Kubernetes Control Plane Restarts on LoadBalancer Service Creation in IPVS Mode

### ISSUE THREAD

**Issue by nberlee** [siderolabs/talos#7159]
### Description
When using kube-proxy in IPVS mode, creating a LoadBalancer service causes a Kubernetes control plane restart. This issue occurs because the LoadBalancer IP assigned to the service is not within the NodeAddressFilter, which leads the Talos controller to recreate certificates and restart the kube-apiserver on all control-plane nodes.

### Reproduce
1. Set up a controller in an environment that can assign a LoadBalancerIP (e.g., MetalLB or a cloud controller).
1. Configure kube-proxy to IPVS mode:
```yaml
cluster:
        proxy:
            mode: ipvs # proxy mode of kube-proxy.
```
3. Create a loadbalancer service: `kubectl create service loadbalancer test --tcp=80` (no container endpoint is needed). Wait until the service is no longer in the "Pending" state for the LoadBalancerIP.
1. Wait a few seconds for kube-proxy to pick up the change on the control plane node(s).
1. Observe the loadBalancerIP in `talosctl get routestatus` on interface kube-ipvs0.
1. Observe `talosctl get nodeaddress`: the loadBalancerIP is considered "accumulative-no-k8s" while the clusterIP is considered "accumulative-only-k8s".
1. Observe the manifests and certificates being changed on the control plane with `talosctl logs controller-runtime`.

### Fix
A temporary fix can be found at https://github.com/nberlee/talos/commit/3b5ff0e21489019e01b5aacc9af8797750c199de, which ignores the kube-ipvs0 interface, I will use this solution for now, until it gets fixed here. 

I will not submit this a PR, as I think a more permanent solution might involve either:

- Allowing the user to specify LoadBalancer subnets in the machine config (not ideal for public IPs in the cloud)
- Or adding kube-ipvs0 /32 IPs that are not in ServiceCIDR to the only-k8s NodeAddressFilter.

### Environment

- Talos version: 1.4.1
- Kubernetes version:1.26.4
- Platform: GCP

**Comment by frezbo** [comment]
good catch, I think i saw the api server restarting when using kube-vip way back, but at that point thought it was a fluke or some issue with kube-vip

**Comment by nberlee** [comment]
Yes it's very hard to pinpoint because of the delay in cloud when assuring a LoadBalancerIP. It has me bit me for a while when flux did his first few runs in a new cluster

This started out as a `flux may overload the kube-apiserver on cluster creation on some clusters' (only when istio was deployed thus cilium was not replacing kube-proxy) to this...

**Comment by frezbo** [comment]
yeh, i thought it was just me, at least we have a bug that's reproducible that's been around for a long while

**Comment by smira** [comment]
I agree this is not the best behavior, even though it should be harmless except for short downtime. 

The problem is how should Talos know that one or another IP is wrong. Another option might be to see if `kube-apiserver` can reload certs without restart.

**Comment by nberlee** [comment]
It is possible in kube-apiserver: with `--encryption-provider-config-automatic-reload`
https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/#configuration-and-determining-whether-encryption-at-rest-is-already-enabled

**Comment by frezbo** [comment]
> It is possible in kube-apiserver: with `--encryption-provider-config-automatic-reload` https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/#configuration-and-determining-whether-encryption-at-rest-is-already-enabled

isn;t this about encryption provider config, and not the actual certs?

**Comment by nberlee** [comment]
> isn;t this about encryption provider config, and not the actual certs?

uh, yes you are right

**Comment by smira** [comment]
Interesting... looking into Kubernetes source code, it feels like it should already support automatic reload

**Comment by nberlee** [comment]
It appears that GCE ILBs rely on Direct Server Return (DSR) as their load balancing mechanism. DSR only modifies the destination MAC address and is utilized during health checks across all VMs within the associated instance groups. This is likely due to K8S being just one of many use cases, and health checks may be conducted on the same port as the load-balanced service. In such cases, the destination IP serves as the distinguishing factor.

**Comment by nberlee** [comment]
Incredible job, @smira! Thank you for fixing it, and backporting it to 1.4, this means I do not have to run my own build anymore.

Now, I am able to confidently move forward with migrating our developers cluster to Talos. In our particular use-case, maintaining an always-available Kubernetes control plane is essential, as our developers rely on filesync to synchronize local files with their pods and monitor logs (using devspace). These commands directly interact with the Kubernetes API, so any control plane restart would have disrupt their workflow.
